snuba/clickhouse/formatter/query.py

```diff
@@ -201,22 +195,29 @@ def _format_arrayjoin(
 
 
 def _format_limit(
-    query: AbstractQuery, formatter: ClickhouseExpressionFormatterBase
+    query: AbstractQuery, formatter: ExpressionVisitor[str]
 ) -> Optional[StringNode]:
     ast_limit = query.get_limit()
     return (
-        StringNode(f"LIMIT {ast_limit} OFFSET {query.get_offset()}")
-        if ast_limit is not None
+        StringNode(f"LIMIT {ast_limit}")
+        if ast_limit is not None and ast_limit != 1000
         else None
     )
 
 
+def _format_offset(
+    query: AbstractQuery, formatter: ExpressionVisitor[str]
+) -> Optional[StringNode]:
+    ast_offset = query.get_offset()
+    return StringNode(f"OFFSET {ast_offset}") if ast_offset != 0 else None
```

Comment:
I've split up the limit and offset format functions, and made it so they don't show up if they're just the default values. Since the formatted queries would only be read by humans, this shouldn't be a problem correct? I can revert the default value check if they should be displayed regardless.

Reply:
I think that's fine.